packages/pluginutils/src/simple-filters.test.ts

```diff
@@ -0,0 +1,65 @@
+// cSpell:ignore fooa, afoo
+import { describe, expect, test } from 'vitest';
+import { exactRegex, prefixRegex } from './simple-filters.js';
+
+describe('exactRegex', () => {
+  test('supports without flag parameter', () => {
+    const regex = exactRegex('foo');
+    expect(regex).toStrictEqual(/^foo$/);
+
+    expect(regex.test('foo')).toBe(true);
+    expect(regex.test('fooa')).toBe(false);
+    expect(regex.test('afoo')).toBe(false);
+  });
+
+  test('supports with flag parameter', () => {
+    const regex = exactRegex('foo', 'i');
+    expect(regex).toStrictEqual(/^foo$/i);
+
+    expect(regex.test('foo')).toBe(true);
+    expect(regex.test('Foo')).toBe(true);
+    expect(regex.test('Fooa')).toBe(false);
+    expect(regex.test('aFoo')).toBe(false);
+  });
+
+  test('escapes special characters for Regex', () => {
+    const regex = exactRegex('foo(bar)');
+    expect(regex).toStrictEqual(/^foo\(bar\)$/);
+
+    expect(regex.test('foo(bar)')).toBe(true);
+    expect(regex.test('foo(bar\\)')).toBe(false);
+    expect(regex.test('foo(bar)a')).toBe(false);
+    expect(regex.test('afoo(bar)')).toBe(false);
+  });
+});
+
+describe('prefixRegex', () => {
+  test('supports without flag parameter', () => {
+    const regex = prefixRegex('foo');
```

Comment:
Would it be better to name this utility to `startsWithRegex` ?

Reply:
I'll prefer `prefixRegex` as it's shorter while expresses the behavior. `startsWith` is easy to understand since `.startsWith` exists, but I think `prefix` is easy enough.

Reply:
make sense